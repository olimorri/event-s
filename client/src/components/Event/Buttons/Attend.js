import React from 'react'
import {
  Button, Popover, PopoverTrigger, Portal, PopoverContent,
  PopoverArrow, PopoverHeader, PopoverCloseButton, PopoverBody, ButtonGroup,
} from '@chakra-ui/react';


export default function Attend ({ handleSubmit }) {

  return (
    <Popover>
      <PopoverTrigger>
        <Button w={'40%'}>Attend</Button>
      </PopoverTrigger>
      <Portal>
        <PopoverContent>
          <PopoverArrow />
          <PopoverHeader>Please confirm!</PopoverHeader>
          <PopoverCloseButton />
          <PopoverBody>
            <ButtonGroup size="sm">
              <Button variant="outline">Cancel</Button>
              <Button onClick={handleSubmit} colorScheme="gray">Attend</Button>
            </ButtonGroup>
          </PopoverBody>
        </PopoverContent>
      </Portal>
    </Popover>
  )
}