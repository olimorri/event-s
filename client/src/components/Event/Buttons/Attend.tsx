import React, { MouseEventHandler } from 'react';
import {
  Button,
  Popover,
  PopoverTrigger,
  Portal,
  PopoverContent,
  PopoverArrow,
  PopoverHeader,
  PopoverCloseButton,
  PopoverBody,
  ButtonGroup,
} from '@chakra-ui/react';

export default function Attend({
  handleClick,
}: {
  handleClick: MouseEventHandler;
}) {
  return (
    <Popover>
      <PopoverTrigger>
        <Button w={'40%'}>Attend</Button>
      </PopoverTrigger>
      <Portal>
        <PopoverContent>
          <PopoverArrow />
          <PopoverHeader>Please confirm!</PopoverHeader>
          <PopoverCloseButton />
          <PopoverBody>
            <ButtonGroup size="sm">
              <Button variant="outline">Cancel</Button>
              <Button onClick={handleClick} colorScheme="gray">
                Attend
              </Button>
            </ButtonGroup>
          </PopoverBody>
        </PopoverContent>
      </Portal>
    </Popover>
  );
}
