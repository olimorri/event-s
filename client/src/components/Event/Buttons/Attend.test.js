import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Attend from './Attend';

const AttendTest = () => (
  <BrowserRouter>
    <Attend />
  </BrowserRouter>
);

describe('Snapshot', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<AttendTest />);
    expect(asFragment(<AttendTest />)).toMatchSnapshot();
  });
});

describe('Unit Testing', () => {
  it('should render the button', () => {
    render(<AttendTest />);
    const attendBtn = screen.getByRole('button', { name: /attend/i });
    expect(attendBtn).toBeInTheDocument();
  });
});
