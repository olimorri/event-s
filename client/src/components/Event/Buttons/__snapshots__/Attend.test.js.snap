// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <button
    aria-controls="popover-content-2"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="chakra-button css-17bcnl0"
    id="popover-trigger-2"
    type="button"
  >
    Attend
  </button>
</DocumentFragment>
`;
